### ISSUE TITLE
Shared folders Vagrant.

### ISSUE THREAD

**Issue by hookercookerman** [fnichol/dvm#7]
do we need guest additions so we can have sync_folder to work?

Cheers

**Comment by fnichol** [comment]
Unfortunately that is correct, which _could_ happen upstream in either https://github.com/mitchellh/boot2docker-vagrant-box or https://github.com/steeve/boot2docker.

However, I'd be curious to know what you'd like to mount into the virtual machine. The only thing that I could come up with was bind mounts to a container?

**Comment by michaelfavia** [comment]
@fnichol: I had the same question but your curiosity about it piqued my curiosity of another possible solution we dont know about.

We were going to use the shared filesystem to allow us to directly edit our html documents in /var/www for docker on our local OSX environments. Perhaps there is another better way?

**Comment by michaelfavia** [comment]
This seems to be being resolved via NSF as the vagrant docs prefer:
http://docs.vagrantup.com/v2/synced-folders/nfs.html

The upstream issue in boot2docker is:
https://github.com/steeve/boot2docker/issues/64

**Comment by tomfitzhenry** [comment]
I have the same use case as @michaelfavia: being able to edit sources in /var/www (for example) in Docker, on local OS X machine.

My current solution to achieve is by not using dvm, and instead using Vagrant/Virtualbox to bring up a Docker-ready host, which will run a container, and have a sync'd folder strewn through VirtualBox (via synced folders) and Docker (via volume binding).

Having the convenience of dvm, rather than configuring my own VirtualBox machine would be handy, however.

**Comment by michaelfavia** [comment]
Last night in a fit of frustration I tried out coreos because it included NFS support. It is a superior alternative to a handrolled vagrant file IMO. Was a breeze to setup. 

I would still much prefer boot2docker (and actually stole a bit of it {dvm env} for my franken setup) and consequently dvm to support NFS but until that happens ill survive. I tried to help upstream with issue 64 but i couldnt get the boot2docker.iso to build properly. Maybe thats my next project.

**Comment by Peeja** [comment]
A use case:

I'm in the process of setting up a Vagrant box for a Rails project which requires running some things that want to be on Linux, like postfix. This is a pretty straightforward use case for Vagrant. The problem is getting the provisioning right. I keep tearing down machines and re-provisioning them.

Through dvm, I get to use a Dockerfile to build an image. If I change the Dockerfile, Docker is smart about doing the least work necessary by reusing intermediate images. If I just need to scrap the container and start fresh, that's super fast in Docker.

In other words, dvm makes a better Vagrant than Vagrant.

Except for sharing files. Vagrant makes that really easy when there's one VM per project, which means I can edit my project on the host and run it on the guest.

If I could mount the source tree from my host Mac in my Vagrant box's Docker container through dvm, that would be pretty sweet.

**Comment by fnichol** [comment]
After revisiting this, the two upstream projects need to support it first before dvm can benefit.

I believe the relevant links are:
- boot2docker/boot2docker#282
- mitchellh/boot2docker-vagrant-box#33

Thanks for the discussion and background.
